micro-frontends/src/next-ui/utils/FormDisplayControl/BuildFormView.js

```diff
@@ -0,0 +1,314 @@
+import axios from "axios";
+import { GET_ALL_FORMS_BASE_URL, GET_FORMS_BASE_URL, GET_FORM_TRANSLATE_URL } from "../../constants";
+import { cloneDeep, remove } from "lodash";
+
+
+var getAllForms = async () => {
+    const apiURL = GET_ALL_FORMS_BASE_URL;
+    const params = {
+        v: "custom:(version,name,uuid)"
+    };
+    try {
+        const response = await axios.get(apiURL, { params });
+        if (response.status === 200) {
+            return response.data;
+        }
+        return [];
+    } catch (error) {
+        console.error(error);
+    }
+};
+
+var getFormDetail = async (formUuid ) => {
+    const apiURL = GET_FORMS_BASE_URL.replace('{formUuid}', formUuid);
+    const params = { 
+        v: "custom:(resources:(value))"
+    };
+    try {
+        const response = await axios.get(apiURL, { params });
+        if (response.status === 200) {
+            return response.data;
+        }
+        return [];
+    } catch (error) {
+        console.error(error);
+    }
+};
+
+
+var getFormTranslate = async (formName, formVersion, locale, formUuid ) => {
+    const apiURL = GET_FORM_TRANSLATE_URL;
+    const params = {
+        "formName": formName,
+        "formVersion": formVersion,
+        "locale": locale,
+        "formUuid": formUuid
+    };
+    try {
+        const response = await axios.get(apiURL, { params });
+        if (response.status === 200) {
+            return response.data;
+        }
+        return [];
+    } catch (error) {
+        console.error(error);
+    }
+};
+var formBuildForms = [];
+export const build = async (bahmniObservations, hasNoHierarchy) => {
+    bahmniObservations.forEach(function (obs) {
+        obs.value = preProcessMultiSelectObs(obs.value);
+    });
+    formBuildForms = await getAllForms();
+    var obs = createObsGroupForForm(bahmniObservations, formBuildForms);
+    if (!hasNoHierarchy) {
```

Comment:
@ayush34957 @SooryaKumaranC-tw Upon reviewing the code, I observed that the build function does not return any response when `hasNoHierarchy` is set to true, and there seems to be no necessity to proceed with the function or network calls that precede this check. Although I lack full context on the card and its requirements, I believe there is potential for enhancing the code structure. 

Adding a conditional check at the start of the build function can optimize the code's efficiency and potentially reduce unnecessary processing. This approach not only improves the code's readability but also ensures better performance by preventing redundant calls when they are not needed. 

cc: @binduak, @angshu

Reply:
@rahu1ramesh The objective is to convert the display control to React. We were making sure to do exact change that was there in angular to react for the forms to get built correctly. We did not try to refactor them because there might be some unknown feature might break if we do so. So we made sure to make minimal changes. You can please refer the existing code too:https://github.com/Bahmni/openmrs-module-bahmniapps/blob/1d4c6c8007913ebc59948e9823b408c0bc93e100/ui/app/common/displaycontrols/observation/services/formRecordTreeBuildService.js#L15